servicesExternal/web3_single.py

```diff
@@ -46,30 +56,60 @@ def arrayToBytes32(self, array, length: int) -> List[Any]:
         pass
 
     def isAddressNoChecksum(self, address: str) -> bool:
-        pass
+        if not address :
+            return False
+        return address and self.web3.utils.isAddress(address.toLowerCase())
 
     def areSameAddressesNoChecksum(self, address1: str, address2: str) -> bool:
-        pass
+        if not address1 or not address2 :
+            return False
+        return address1.toLowerCase()==address2.toLowerCase()
 
     def isHexStrictBytes32(self, hex: str) -> bool:
-        pass
+        return self.web3.utils.isHexStrict(hex) and hex.length == 66
 
     def generateWeb3Method(self, contractInstance,
                            name: str,
                            parameters: List[Any]) -> Any:
-        pass
+        return contractInstance.methods[name].apply(None, parameters)
 
-    def decodeInputData(self, abi: Any, data: str) -> Any:
+    def decodeInputData(self, abi: List[Any], data: str) -> Any:
         pass
 
     def decodeTransactionLog(self, abi: List[Any], event: str, log: Any) -> Any:
-        pass
+        'eventInput' : Any
```

Comment:
Should just be `eventInput : Any`, no need for the `'`s. Same for the one below.